flips/20220503-interaction-templates.md

```diff
@@ -0,0 +1,446 @@
+                    zh-CN: "将 {amount} FLOW 转移到 {to}"
+                }
+            }
+        },
+        cadence: // Cadence code this interaction executes.
+        ` 
+        import FungibleToken from 0xFUNGIBLETOKENADDRESS
+        transaction(amount: UFix64, to: Address) {
+            let vault: @FungibleToken.Vault
+            prepare(signer: AuthAccount) {
+                %%self.vault <- signer
+                .borrow<&{FungibleToken.Provider}>(from: /storage/flowTokenVault)!
+                .withdraw(amount: amount)%%ount)
+            
+                self.vault <- FungibleToken.getVault(signer)
+            }
+            execute {
+                getAccount(to)
+                .getCapability(/public/flowTokenReceiver)!
+                .borrow<&{FungibleToken.Receiver}>()!
+                .deposit(from: <-self.vault)
+            }
+        }
+        `,
+        dependencies: {
+            "0xFUNGIBLETOKENADDRESS": { // Network (mainnet || testnet) dependent locations for 0xFUNGIBLETOKENADDRESS contract.
+                "FungibleToken" : {
+                    mainnet: {
+                        address: "0xf233dcee88fe0abe", // Address of the account the contract is located.
+                        fq_address: "A.0xf233dcee88fe0abe.FungibleToken", // Fully qualified contract identifier.
+                        pin: "asdfasdfasdfasdfasdfasdfsadf123123123123", // Unique identifier of the interactions dependency tree.
+                        pin_block_height: 10123123123 // Block height the pin was generated against.
+                    },
+                    testnet: {
+                        address: "0x9a0766d93b6608b7",
+                        fq_address: "A.0x9a0766d93b6608b7.FungibleToken",
+                        pin: "asdfasdfasdfasdfasdfasdfsadf123123123123",
+                        pin_block_height: 10123123123
+                    }
+                }
+            }
+        },
+        arguments: {
+            amount: {
+                index: 0,
+                messages: { // Set of human readable messages about the argument
+                    title: {
+                        i18n: { // Internationalised (BCP-47) set of human readable messages about the argument
+                            en-US: "Amount",
+                            fr-FR: "Montant",
+                            zh-CN: "数量",
+                        }
+                    },
+                    description: {
+                        i18n: { // Internationalised (BCP-47) set of human readable messages about the argument
+                            en-US: "Amount of FLOW token to transfer",
+                            fr-FR: "Quantité de token FLOW à transférer",
+                            zh-CN: "要转移的 FLOW 代币数量"
+                        }
+                    }
+                },
+                balance: "0xFUNGIBLETOKENADDRESS.FungibleToken" // The token this argument acts upon.
```

Comment:
`balance` seems to not fit here, would all arguments have "balance" property?

Reply:
No not all arguments would have a "balance" property. What this aims to do is create a link between the dependencies of the interaction and the argument to show they're related. With this property an app/wallet might be able to use it to determine if a user has sufficient balance of a FT to successfully execute a transaction, for example.